js_assignment4.js

```diff
@@ -0,0 +1,41 @@
+const SortArray =function (orignalArray){
+    this.temp_array=orignalArray;
+}
+SortArray.prototype.getSortedArray=function(){
+    let new_array=this.temp_array.sort((a,b)=> a-b);
+    return new_array ;
+};
+//const  arry= new SortArray([2,3,1,4]);
+//console.log(arry.getSortedArray());
+
+const SortObjectArray = function(orignalArray){
+  SortArray.call(this,orignalArray);
+}
+SortObjectArray.prototype=Object.create(SortArray.prototype);
+
+SortObjectArray.prototype.getSortedObjectArray=function(key){
+  let new_array=[];
```

Comment:
Can we have the same name for the sorting function and based on the object we should dynamically call the sorting function?